src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Create.cs

```diff
@@ -53,28 +42,32 @@ internal sealed partial class SyntaxTreeIndex
         /// this string table.  The table will have already served its purpose at that point and 
         /// doesn't need to be kept around further.
         /// </summary>
-        private static readonly ConditionalWeakTable<Project, StringTable> s_projectStringTable =
-            new();
+        private static readonly ConditionalWeakTable<Project, StringTable> s_projectStringTable = new();
 
         private static async Task<SyntaxTreeIndex> CreateIndexAsync(
             Document document, Checksum checksum, CancellationToken cancellationToken)
         {
-            var project = document.Project;
-            var stringTable = GetStringTable(project);
+            Contract.ThrowIfFalse(document.SupportsSyntaxTree);
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
pulled this await up, and put everything else in a synchronous method to prevent pooled collections from being used across awaits (As per @sharwell ).